internal/app/subsystems/api/http/http.go

```diff
@@ -2,28 +2,22 @@ package http
 
 import (
 	"context"
+	"github.com/resonatehq/resonate/internal/creds"
 	"net/http"
 	"time"
 
 	"github.com/go-playground/validator/v10"
 	"github.com/resonatehq/resonate/internal/app/subsystems/api/service"
-	"github.com/resonatehq/resonate/internal/util"
-
 	"log/slog"
 
 	"github.com/gin-gonic/gin"
 	"github.com/gin-gonic/gin/binding"
 	"github.com/resonatehq/resonate/internal/api"
 )
 
-type Auth struct {
-	Username string
-	Password string
-}
-
 type Config struct {
 	Addr    string
-	Auth    *Auth
+	Auth    []interface{}
```

Comment:
```suggestion
	Auth    map[string]string
```

I was doing some experimenting and I think using the the map type that gin expects will make our life a lot easier :)

Reply:
In order to do this we need to make two small tweaks to the flags defined in `cmd/serve/serve.go`. We need to replace these flags:
- api-http-auth-username / api.subsystems.http.auth.username
- api-http-auth-password / api.subsystems.http.auth.password

With a single map flag
```go
cmd.Flags().StringToString("api-http-auth", map[string]string{}, "basic auth username/password pairs")
_ = viper.BindPFlag("api.subsystems.http.auth", cmd.Flags().Lookup("api-http-auth"))
```

Reply:
Yup, changed the type as indeed we would need to pass a map to gin , and added cobra flags as suggested by you , as i undertsand we do also have to keep the option to provide credentials via the command line too. Also we need to bind  with viper because viper needs to be the single source of truth for 'getting' almost all values, both from a config file as well as those that get passed on command line :-)